src/Modules/Legacy/Inverse/Tests/TikhonovFunctionalTest.cc

```diff
@@ -185,3 +194,585 @@ TEST_F(TikhonovFunctionalTest, DISABLED_loadIDNonSquareFwdMatrixANDRandDataDiffS
+    EXPECT_THROW(tikAlgImp->execute(), SCIRun::Core::LinearAlgebra::LapackError);
+}
+
+
+// ID forward matrix with source regularization matrix and single lambda
+TEST_F(TikhonovFunctionalTest, functionTestIDFwdMatrixANDRandDataInputIDSourceReguSingleLambda)
+{
+    // create inputs
+    auto tikAlgImp = makeModule("SolveInverseProblemWithTikhonov");
+    MatrixHandle fwdMatrix(new DenseMatrix(DenseMatrix::Identity(3, 3)));    // forward matrix (IDentityt)
+    MatrixHandle measuredData(new DenseMatrix(DenseMatrix::Random(3, 1)));   // measurement data (rand)
+    MatrixHandle sourceRegularizationMatrix(new DenseMatrix(DenseMatrix::Zero(3, 3)));    // source regularization matrix (identity)
+    MatrixHandle inverseSolution;
+    DenseMatrix residual;
+    
+    // input data
+    stubPortNWithThisData(tikAlgImp, 0, fwdMatrix);
+    stubPortNWithThisData(tikAlgImp, 2, measuredData);
+    stubPortNWithThisData(tikAlgImp, 1, sourceRegularizationMatrix);
+    
+    // change params
+    tikAlgImp->setStateDefaults();                                                  // set default params
+    tikAlgImp->get_state()->setValue(SolveInverseProblemWithTikhonov::RegularizationMethod, std::string("single"));  // select single lambda
+    tikAlgImp->get_state()->setValue(SolveInverseProblemWithTikhonov::LambdaFromDirectEntry, 1 );                    // change lambda
+    
+    // execute
+    tikAlgImp->execute();
+    
+    inverseSolution = getDataOnThisOutputPort(tikAlgImp,0);
+    residual = *inverseSolution - measuredData;
+    
+    // check result
+    ASSERT_NEAR( residual.norm(), 0,  abs_error );
+
+}
+
+// ID forward matrix with source regularization matrix and default L-curve
+TEST_F(TikhonovFunctionalTest, functionTestIDFwdMatrixANDRandDataInputIDSourceReguSingleLambda)
+{
+    // create inputs
+    auto tikAlgImp = makeModule("SolveInverseProblemWithTikhonov");
+    MatrixHandle fwdMatrix(new DenseMatrix(DenseMatrix::Identity(3, 3)));    // forward matrix (IDentityt)
+    MatrixHandle measuredData(new DenseMatrix(DenseMatrix::Random(3, 1)));   // measurement data (rand)
+    MatrixHandle sourceRegularizationMatrix(new DenseMatrix(DenseMatrix::Zero(3, 3)));    // source regularization matrix (identity)
+    MatrixHandle inverseSolution;
+    DenseMatrix residual;
+    
+    // input data
+    stubPortNWithThisData(tikAlgImp, 0, fwdMatrix);
+    stubPortNWithThisData(tikAlgImp, 2, measuredData);
+    stubPortNWithThisData(tikAlgImp, 1, sourceRegularizationMatrix);
+    
+    // change params -> DEFAULT
+    
+    // execute
+    tikAlgImp->execute();
+    
+    inverseSolution = getDataOnThisOutputPort(tikAlgImp,0);
+    residual = *inverseSolution - measuredData;
+
+    
+    // check result
+    ASSERT_NEAR( residual.norm(), 0,  abs_error );
+    
+}
+
+
+// ID forward matrix with residual regularization matrix and single lambda
+TEST_F(TikhonovFunctionalTest, functionTestIDFwdMatrixANDRandDataInputIDResidualReguWithSingleLambda)
```

Comment:
@jcollfont Rather than leave commented-out test code, if you can't get a test working, just add `DISABLED_` to the start of the test name and it won't be run as part of the suite. This will keep the code from getting stale.